libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -523,7 +523,7 @@ void Compass::_detect_backends(void)
 #if CONFIG_HAL_BOARD == HAL_BOARD_PX4
     if (AP_BoardConfig::get_board_type() == AP_BoardConfig::PX4_BOARD_PIXHAWK2) {
         // default to disabling LIS3MDL on pixhawk2 due to hardware issue
-        _driver_type_mask.set_default(1U<<DRIVER_LIS3MDL);
+        _driver_type_mask.set_default(1U<<DRIVER_LIS3MDL | 1U<<DRIVER_QMC5883);
```

Comment:
As far as I understood the problem is with the driver, not with the interaction with that board. Why are we disabling it only on pixhawk2?